### PR TITLE
frieren: lr-min cosine floor — prevent schedule decay to zero

### PR DESCRIPTION
**Pull request by morganmcg1**
# frieren: lr-min cosine floor — prevent schedule decay to zero

## Hypothesis

The current SOTA stack uses `--lr-cosine-t-max 13` matched to `--epochs 13`, which means the cosine schedule decays the learning rate all the way to **zero** at the final epoch. This may suppress learning of the high-frequency STRING-separable positional encoding parameters (RFF sigmas, frequency bands) precisely when the model is trying to refine its hardest predictions — the τ_y/τ_z components.

Setting `--lr-min 1e-5` (one order of magnitude below the peak lr=1e-4) places a non-zero floor under the cosine decay so the optimizer continues to make meaningful updates through EP12–EP13. Lion in particular benefits from a non-collapsing lr because its update magnitude is sign-based and a near-zero lr produces near-zero updates regardless of gradient signal.

This is an orthogonal hyperparameter that has **never been tested in any prior PR** (verified via repo search). It is a low-complexity change with a plausible mechanism and zero risk of breaking the SOTA stack.

## Background

- Current SOTA: PR #510 alphonse, val_abupt=**7.0063%**, test_abupt=**8.2921%** (W&B `qqtdnlwq`).
- Per-axis val: sp=4.5994%, ws=7.8939%, vp=4.1643%, τ_x=6.8150%, τ_y=**8.9516%**, τ_z=**10.5010%**.
- Primary research target: close the τ_y / τ_z gap (SOTA 8.95%/10.50% vs AB-UPT paper 3.65%/3.63%).
- `--lr-min` is a CLI flag on `train.py` already; default is 0.0 (used in every prior run).
- None of the 6 in-flight tay WIP PRs (#516, #523, #534, #536, #537, #538, #549) touch the lr schedule floor.

## Experiment

Single training run on the full SOTA stack with one change: `--lr-min 1e-5`.

```bash
torchrun --standalone --nproc_per_node=8 train.py \
  --agent frieren --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --vol-points-schedule "0:16384:3:32768:6:49152:9:65536" \
  --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable --use-qk-norm \
  --rff-num-features 16 \
  --rff-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --lr-cosine-t-max 13 \
  --epochs 13 \
  --surface-loss-weight 2.0 \
  --lr-min 1e-5 \
  --wandb-group frieren-lr-min-floor --wandb-name frieren/lr-min-1e-5
```

Only difference from SOTA reproduce: `--lr-min 1e-5` (added).

## Success criteria

- **Win**: val_abupt < 7.0063% (any improvement; merge-eligible).
- **Strong win**: τ_y or τ_z drops below SOTA values (8.9516% / 10.5010%).
- **Interesting**: late-epoch (EP10–13) per-axis val errors continue to decrease rather than plateauing — signal that the floor is doing real work.
- **Null**: matches SOTA within noise → close, log that `--lr-min` does not help with this peak lr / schedule combo.

## Reporting

Add a comment on this PR with:
- Final val and test ABUPT (per-axis breakdown: sp, ws, vp, τ_x, τ_y, τ_z).
- W&B run ID and group link.
- Per-epoch val_abupt curve for EP1–EP13 (especially EP10–13 to see if the floor matters).
- Lion update-norm or grad-norm in late epochs vs SOTA run, if available.
- One-line conclusion: did the floor help, hurt, or no-op?

Then `mark_ready_for_review`.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR heads-up: The branch `frieren/surface-vol-sigma-split` is mergeable-conflicting against `tay` (advisor branch has moved since this assignment was opened). Before launching the experiment, please:

1. `git fetch origin tay && git rebase origin/tay`
2. Resolve any conflicts (likely just BASELINE.md / EXPERIMENTS_LOG.md)
3. Push the rebased branch and proceed with the run

Note: branch name mentions `surface-vol-sigma-split` but the experiment described in the title/body is `--lr-min 1e-5` cosine floor. The lr-min experiment is what we want — ignore the old branch name. If you want to rename the branch for clarity that's fine but not required.

Also FYI: SOTA has moved since this PR was authored. Current baseline is **PR #523 thorfinn EMA-proxy GradNorm, val 6.9246% / test 8.2355%** (run `wyz68o8r`), not PR #510. And askeladd PR #516 (currently rebasing) reported val 6.8701% / test 8.1229% which will land soon. Adjust your success criteria accordingly:
- **Win:** val_abupt < 6.9246% (or whichever SOTA is on `tay` when you start)
- **Strong win:** test_abupt beats current `tay` test SOTA

The lr-min hypothesis is still well-motivated — a non-zero floor on the SOTA stack hasn't been tested. Add `--use-gradnorm --gradnorm-mode ema_proxy --gradnorm-alpha 0.5 --gradnorm-min-weight 0.7` to your reproduce command to match the new (PR #523) SOTA stack.

**Comment by morganmcg1** [comment]
STUDENT-WATCHDOG: stopping this student invocation on senpai-drivaerml-ddp8-frieren-776db5df79-r4z9q.

Reason: assignment changed from #550 to none for 1777847176s while train.py is active
Original assignment: #550
Current assignment: none

Active training processes before termination:

```
 242577       1       2  371  0.0 pt_elastic      /usr/bin/python3 /usr/local/bin/torchrun --standalone --nproc_per_node=8 train.py --agent frieren --optimizer lion --lr 1e-4 --weight-decay 5e-4 --no-compile-model --batch-size 4 --validation-every 1 --train-surface-points 65536 --eval-surface-points 65536 --train-volume-points 65536 --eval-volume-points 65536 --vol-points-schedule 0:16384:3:32768:6:49152:9:65536 --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 --pos-encoding-mode string_separable --use-qk-norm --rff-num-features 16 --rff-init-sigmas 0.25,0.5,1.0,2.0,4.0 --lr-cosine-t-max 13 --epochs 13 --use-gradnorm --gradnorm-mode ema_proxy --gradnorm-alpha 0.5 --gradnorm-min-weight 0.7 --gradnorm-ema-beta 0.9 --gradnorm-init-warmup-steps 50 --lr-min 1e-5 --wandb-group frieren-lr-min-floor --wandb-name frieren/lr-min-1e-5
 242646  242577       0 97.7  0.0 python3         /usr/bin/python3 -u train.py --agent frieren --optimizer lion --lr 1e-4 --weight-decay 5e-4 --no-compile-model --batch-size 4 --validation-every 1 --train-surface-points 65536 --eval-surface-points 65536 --train-volume-points 65536 --eval-volume-points 65536 --vol-points-schedule 0:16384:3:32768:6:49152:9:65536 --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 --pos-encoding-mode string_separable --use-qk-norm --rff-num-features 16 --rff-init-sigmas 0.25,0.5,1.0,2.0,4.0 --lr-cosine-t-max 13 --epochs 13 --use-gradnorm --gradnorm-mode ema_proxy --gradnorm-alpha 0.5 --gradnorm-min-weight 0.7 --gradnorm-ema-beta 0.9 --gradnorm-init-warmup-steps 50 --lr-min 1e-5 --wandb-group frieren-lr-min-floor --wandb-name frieren/lr-min-1e-5
 242647  242577       0 97.7  0.0 python3         /usr/bin/python3 -u train.py --agent frieren --optimizer lion --lr 1e-4 --weight-decay 5e-4 --no-compile-model --batch-size 4 --validation-every 1 --train-surface-points 65536 --eval-surface-points 65536 --train-volume-points 65536 --eval-volume-points 65536 --vol-points-schedule 0:16384:3:32768:6:49152:9:65536 --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 --pos-encoding-mode string_separable --use-qk-norm --rff-num-features 16 --rff-init-sigmas 0.25,0.5,1.0,2.0,4.0 --lr-cosine-t-max 13 --epochs 13 --use-gradnorm --gradnorm-mode ema_proxy --gradnorm-alpha 0.5 --gradnorm-min-weight 0.7 --gradnorm-ema-beta 0.9 --gradnorm-init-warmup-steps 50 --lr-min 1e-5 --wandb-group frieren-lr-min-floor --wandb-name frieren/lr-min-1e-5
 242648  242577       0 97.7  0.0 python3         /usr/bin/python3 -u train.py --agent frieren --optimizer lion --lr 1e-4 --weight-decay 5e-4 --no-compile-model --batch-size 4 --validation-every 1 --train-surface-points 65536 --eval-surface-points 65536 --train-volume-points 65536 --eval-volume-points 65536 --vol-points-schedule 0:16384:3:32768:6:49152:9:65536 --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 --pos-encoding-mode string_separable --use-qk-norm --rff-num-features 16 --rff-init-sigmas 0.25,0.5,1.0,2.0,4.0 --lr-cosine-t-max 13 --epochs 13 --use-gradnorm --gradnorm-mode ema_proxy --gradnorm-alpha 0.5 --gradnorm-min-weight 0.7 --gradnorm-ema-beta 0.9 --gradnorm-init-warmup-steps 50 --lr-min 1e-5 --wandb-group frieren-lr-min-floor --wandb-name frieren/lr-min-1e-5
 242649  242577       0  100  0.0 python3         /usr/bin/python3 -u train.py --agent frieren --optimizer lion --lr 1e-4 --weight-decay 5e-4 --no-compile-model --batch-size 4 --validation-every 1 --train-surface-points 65536 --eval-surface-points 65536 --train-volume-points 65536 --eval-volume-points 65536 --vol-points-schedule 0:16384:3:32768:6:49152:9:65536 --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 --pos-encoding-mode string_separable --use-qk-norm --rff-num-features 16 --rff-init-sigmas 0.25,0.5,1.0,2.0,4.0 --lr-cosine-t-max 13 --epochs 13 --use-gradnorm --gradnorm-mode ema_proxy --gradnorm-alpha 0.5 --gradnorm-min-weight 0.7 --gradnorm-ema-beta 0.9 --gradnorm-init-warmup-steps 50 --lr-min 1e-5 --wandb-group frieren-lr-min-floor --wandb-name frieren/lr-min-1e-5
 242650  242577       0  100  0.0 python3         /usr/bin/python3 -u train.py --agent frieren --optimizer lion --lr 1e-4 --weight-decay 5e-4 --no-compile-model --batch-size 4 --validation-every 1 --train-surface-points 65536 --eval-surface-points 65536 --train-volume-points 65536 --eval-volume-points 65536 --vol-points-schedule 0:16384:3:32768:6:49152:9:65536 --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 --pos-encoding-mode string_separable --use-qk-norm --rff-num-features 16 --rff-init-sigmas 0.25,0.5,1.0,2.0,4.0 --lr-cosine-t-max 13 --epochs 13 --use-gradnorm --gradnorm-mode ema_proxy --gradnorm-alpha 0.5 --gradnorm-min-weight 0.7 --gradnorm-ema-beta 0.9 --gradnorm-init-warmup-steps 50 --lr-min 1e-5 --wandb-group frieren-lr-min-floor --wandb-name frieren/lr-min-1e-5
 242651  242577       0  100  0.0 python3         /usr/bin/python3 -u train.py --agent frieren --optimizer lion --lr 1e-4 --weight-decay 5e-4 --no-compile-model --batch-size 4 --validation-every 1 --train-surface-points 65536 --eval-surface-points 65536 --train-volume-points 65536 --eval-volume-points 65536 --vol-points-schedule 0:16384:3:32768:6:49152:9:65536 --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 --pos-encoding-mode string_separable --use-qk-norm --rff-num-features 16 --rff-init-sigmas 0.25,0.5,1.0,2.0,4.0 --lr-cosine-t-max 13 --epochs 13 --use-gradnorm --gradnorm-mode ema_proxy --gradnorm-alpha 0.5 --gradnorm-min-weight 0.7 --gradnorm-ema-beta 0.9 --gradnorm-init-warmup-steps 50 --lr-min 1e-5 --wandb-group frieren-lr-min-floor --wandb-name frieren/lr-min-1e-5
 242652  242577       0 97.7  0.0 python3         /usr/bin/python3 -u train.py --agent frieren --optimizer lion --lr 1e-4 --weight-decay 5e-4 --no-compile-model --batch-size 4 --validation-every 1 --train-surface-points 65536 --eval-surface-points 65536 --train-volume-points 65536 --eval-volume-points 65536 --vol-points-schedule 0:16384:3:32768:6:49152:9:65536 --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 --pos-encoding-mode string_separable --use-qk-norm --rff-num-features 16 --rff-init-sigmas 0.25,0.5,1.0,2.0,4.0 --lr-cosine-t-max 13 --epochs 13 --use-gradnorm --gradnorm-mode ema_proxy --gradnorm-alpha 0.5 --gradnorm-min-weight 0.7 --gradnorm-ema-beta 0.9 --gradnorm-init-warmup-steps 50 --lr-min 1e-5 --wandb-group frieren-lr-min-floor --wandb-name frieren/lr-min-1e-5
 242653  242577       0  100  0.0 python3         /usr/bin/python3 -u train.py --agent frieren --optimizer lion --lr 1e-4 --weight-decay 5e-4 --no-compile-model --batch-size 4 --validation-every 1 --train-surface-points 65536 --eval-surface-points 65536 --train-volume-points 65536 --eval-volume-points 65536 --vol-points-schedule 0:16384:3:32768:6:49152:9:65536 --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 --pos-encoding-mode string_separable --use-qk-norm --rff-num-features 16 --rff-init-sigmas 0.25,0.5,1.0,2.0,4.0 --lr-cosine-t-max 13 --epochs 13 --use-gradnorm --gradnorm-mode ema_proxy --gradnorm-alpha 0.5 --gradnorm-min-weight 0.7 --gradnorm-ema-beta 0.9 --gradnorm-init-warmup-steps 50 --lr-min 1e-5 --wandb-group frieren-lr-min-floor --wandb-name frieren/lr-min-1e-5
```

The watchdog only takes this action after the assignment drift remains stable for 1800s, so transient GitHub/API wobbles should not trigger it.
